app/cf.py

```diff
@@ -0,0 +1,59 @@
+import os
+import time
+import logging
+
+from cloudfoundry_client.client import CloudFoundryClient
+
+
+class Cf:
+    def __init__(self):
+        self.client = None
+
+        self.org = os.environ['CF_ORG']
+        self.space = os.environ['CF_SPACE']
+        self.api_url = os.environ['CF_API_URL']
+        self.username = os.environ['CF_USERNAME']
+        self.password = os.environ['CF_PASSWORD']
+
+    def get_cloudfoundry_client(self):
+        if self.client is None:
+            proxy = dict(http=os.environ.get('HTTP_PROXY', ''), https=os.environ.get('HTTPS_PROXY', ''))
+            client = CloudFoundryClient(self.api_url, proxy=proxy)
+            try:
+                client.init_with_user_credentials(self.username, self.password)
+                self.client = client
+            except BaseException as e:
+                msg = 'Failed to authenticate: {}, waiting 5 minutes and exiting'.format(str(e))
+                logging.error(msg)
+                # The sleep is added to avoid automatically banning the user for too many failed login attempts
+                time.sleep(5 * 60)
```

Comment:
Maybe one for the backlog: Replace thread.sleep with something else e.g.

```
from retrying import retry

@retry
def logon():
    try:
        client.init_with_user_credentials(self.username, self.password)
   except BaseException as e:
        raise Exception("Log in failed")
```

There are parameters which can do the retry times and delay etc

`@retry(wait_exponential_multiplier=1000, wait_exponential_max=10000, stop_max_delay=30000)`